tests/acceptance/30_custom_promise_types/bodies.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/python3
+#
+# Sample custom promise type, uses cfengine.py library located in same dir.
+#
+# Use it in the policy like this:
+# promise agent append
+# {
+#   interpreter => "/usr/bin/python3";
+#   path => "$(sys.inputdir)/bodies.py";
+# }
+# body example_type example_name
+# {
+#   foo => "Hello";
+#   bar => "World";
+# }
+# bundle agent main
+# {
+#   append:
+#       "/path/to/target/file"
+#         example_type => example_name;
+# }
+
+
+import json
+from cfengine import PromiseModule, ValidationError, Result
+
+
+class BodiesPromiseTypeModule(PromiseModule):
+    def __init__(self):
+        super().__init__("bodies_promise_module", "0.0.1")
+
+    def validate_promise(self, promiser, attributes):
+        pass
+
+    def evaluate_promise(self, promiser, attributes):
+        try:
+            with open(promiser, "r") as f:
+                if (json.dumps(attributes) == f.read()):
+                    self.log_verbose("Promise '%s' kept" % promiser)
+                    return Result.KEPT
+                f.write(json.dumps(attributes))
+        except:
+            try:
+                with open(promiser, "w") as f:
+                    f.write(json.dumps(attributes))
+                    self.log_verbose("Promise '%s' repaired" % promiser)
```

Comment:
This should be `self.log_info()`. Not important here, but you never know when somebody takes this as a template for their module.

Reply:
@larsewi @vpodzime no, we've changed / fixed this in other places. If you are just writing the outcome of the promise it should be verbose. But there should be a INFO message about what change was made to the system.

Reply:
Right.

Reply:
So the line should actually be dropped or it should be something like:
```
self.log_info("Written attributes into file %s" % promiser)
```